src/schema/v2/artwork/collectorSignals.ts

```diff
@@ -29,48 +27,267 @@ export const CollectorSignals: GraphQLFieldConfig<any, ResolverContext> = {
       bidCount: {
         type: GraphQLInt,
         description: "Bid count on lots open for bidding",
+        resolve: async (artwork, {}, ctx) => {
+          const isInSale = artwork.sale_ids?.length > 0
+
+          if (
+            !checkFeatureFlag("emerald_signals-auction-improvements", ctx) ||
+            !isInSale
+          ) {
+            return null
+          }
+
+          const activeLotData = await getActiveSaleArtwork(
+            {
+              artworkId: artwork.id,
+              saleIds: artwork.sale_ids,
+            },
+            ctx
+          )
+
+          if (activeLotData) {
+            return activeLotData.saleArtwork.bidder_positions_count
+          }
+
+          return null
+        },
       },
       lotWatcherCount: {
         type: GraphQLInt,
         description: "Lot watcher count on lots open for bidding",
+        resolve: async (artwork, {}, ctx) => {
+          const isInSale = artwork.sale_ids?.length > 0
+
+          if (
+            !checkFeatureFlag("emerald_signals-auction-improvements", ctx) ||
+            !isInSale
+          ) {
+            return null
```

Comment:
Same comment as above - maybe all the auction-specific stuff goes into `auction` under `collectorSignals`? Then the actual numeric fields can all be non-null, plus you wouldn't necessarily repeat this code for all the auction-specific signals.